ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -451,31 +455,86 @@ EOF
   done
 }
 
-# inject_spot_instance_config is an AWS specific option that enables the use of AWS spot instances for worker nodes
+# inject_spot_instance_config is an AWS specific option that enables the
+# use of AWS spot instances.
+# PARAMS:
+# $1: Path to base output directory of `openshift-install create manifests`
+# $2: Either "workers" or "masters" to enable spot instances on the
+#     compute or control machines, respectively.
 function inject_spot_instance_config() {
   local dir=${1}
+  local mtype=${2}
+
+  get_yq
+
+  # Find manifest files
+  local manifests=
+  case "${mtype}" in
+    masters)
+      # Spot masters works for
+      # - CAPA, always
+      # - Terraform, only for newer installer binaries containing https://github.com/openshift/installer/pull/8349
+      if [[ "$(/tmp/yq r "${SHARED_DIR}/install-config.yaml" 'featureGates[*]')" =~ ClusterAPIInstall(AWS)?=true ]]; then
```

Comment:
CAPA based installs are now default so these feature gates will not be set for this case. CAPA based installs could be identified by the present of the CAPI manifests dir

Reply:
Okay, fixed.

So there will be another PR to remove the feature flags from all those jobs? :innocent: